cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/service/subnetselector/SubnetSelectorService.java

```diff
@@ -0,0 +1,78 @@
+package com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+
+@Service
+class SubnetSelectorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubnetSelectorService.class);
+
+    Map<String, CloudSubnet> collectOnePrivateSubnetPerAz(List<CloudSubnet> subnetMetas, int max) {
+        Map<String, CloudSubnet> subnetsPerAz = new HashMap<>();
+        Iterator<CloudSubnet> subnetIterator = subnetMetas.iterator();
+        while (subnetsPerAz.size() < max && subnetIterator.hasNext()) {
+            CloudSubnet nextSubnet = subnetIterator.next();
+            if (nextSubnet.isPrivateSubnet()) {
+                subnetsPerAz.putIfAbsent(nextSubnet.getAvailabilityZone(), nextSubnet);
+            }
+        }
```

Comment:
this could be simplified with stream, like:
`subnetMetas.stream().filter(isPrivate).limit(max).collect(Collectors.toMap)`

Reply:
Thank you for the suggestion, did it.

Reply:
Hm, it does not work. I think limit will stop checking for more private subnets when it reached the desired count, irrespective of their AZ.